pkg/webhook/cspc_operations.go

```diff
@@ -327,41 +285,43 @@ func (bdr *BlockDeviceReplacement) IsNewBDPresentOnCurrentCSPC(newRG, oldRG *cst
 }
 
 // IsExistingReplacmentInProgress returns true if a block device in raid group is under active replacement.
-func (bdr *BlockDeviceReplacement) IsExistingReplacmentInProgress(oldRG *cstor.RaidGroup) (bool, error) {
+func (pOps *PoolOperations) IsExistingReplacmentInProgress(oldRG *cstor.RaidGroup) (bool, error) {
 	for _, v := range oldRG.CStorPoolInstanceBlockDevices {
-		bdcObject, err := bdr.GetBDCOfBD(v.BlockDeviceName)
+		bdcObject, err := pOps.GetBDCOfBD(v.BlockDeviceName)
 		if err != nil {
 			return true, errors.Errorf("failed to query for any existing replacement in the raid group : %s", err.Error())
 		}
-		_, ok := bdcObject.GetAnnotations()[types.PredecessorBDLabelKey]
-		if ok {
-			return true, errors.Errorf("replacement is still in progress for bd %s", v.BlockDeviceName)
+		if bdcObject != nil {
```

Comment:
This can be improved by returning a typed error -- like BDC does not exists. 
And BDC does not exist --  should be done instead. 
Philosophy is -- if err is nil -- object should be not nil -- code looks more cleaner.
Commenting for tracking. It can be enhanced in a new PR.

Reply:
Anyways -- thanks. Nice Catch.